File: go/array/1128. 等价多米诺骨牌对的数量.go
package array

func numEquivDominoPairs(dominoes [][]int) int {
	nums := make([]int, 100)
	var res int
	for i, _ := range dominoes {
		x, y := dominoes[i][0], dominoes[i][1]
		var val int
		if x < y {
			val = x*10 + y
		} else {
			val = y*10 + x
		}
		res += nums[val]
		nums[val]++
	}
	return res
}

File: python/数组/1128. 等价多米诺骨牌对的数量.py
from typing import List
class Solution:
    def numEquivDominoPairs(self, dominoes: List[List[int]]) -> int:
        nums = [0] * 100
        res = 0
        for x,y in dominoes:
            val = x*10+y if x < y else y*10+x
            res += nums[val]
            nums[val]+=1
        return res

